pkg/k8s/state_into_spec_annotation.go

```diff
@@ -24,12 +24,12 @@ import (
 // ValidateOrDefaultStateIntoSpecAnnotation validates the value of the
 // 'state-into-spec' annotation if it is set and defaults the annotation to the
 // passed in defaultValue if it is unset.
-func ValidateOrDefaultStateIntoSpecAnnotation(obj client.Object, defaultValue string) error {
+func ValidateOrDefaultStateIntoSpecAnnotation(obj client.Object, defaultValue string) (defaulted bool, err error) {
```

Comment:
Nit: The fn name seems to be a little bit redundant and ambiguous.  You may want to consider restructuring the code to make this a struct function of `StateIntoSpecDefaulter`. i.e. 
```go
func (*StateIntoSpecDefaulter) ApplyDefaults(..) {...}

func (*StateIntoSpecDefaulter) validate() err { # previous validateStateIntoSpecAnnotation}
func (*StateIntoSpecDefaulter) setEmpty( ) bool {# kind of like the previous setStateIntoSpecDefaultValueIfAllowed }
```

Reply:
Did some refactoring to merge the logic in to ApplyDefaults().

I didn't make other functions the methods for the StateIntoSpecDefaulter because I'd expect validation to be needed in the future in webhook for the turndown. A defaulter is not required when we do the annotation value checks like that.